lib/smack/SmackInstGenerator.cpp

```diff
@@ -166,18 +166,35 @@ void SmackInstGenerator::processIntrinsicCall(llvm::IntrinsicInst* ii) {
       funcName = "__SMACK_convert_to_fp16_f64";
     else
       llvm_unreachable("Unexpected operand type of half-precision downcast intrinsic.");
-    return Stmt::call(funcName, {rep->expr(ii->getArgOperand(0))},
-      {naming->get(*ii)});
+    emit(Stmt::call(funcName, {rep->expr(ii->getArgOperand(0))},
+      {naming->get(*ii)}));
   };
-  static const std::map<llvm::Intrinsic::ID, std::function<const Stmt*()>> stmtMap {
+  static const auto expect = [&ii, this] {
+    // The llvm.expect.* function has two arguments: a value v and an expected
+    // value that is supposed to be used by optimizers.
+    // Semantically, this function simply returns the value v.
+    Value* val = ii->getArgOperand(0);
+    emit(Stmt::assign(rep->expr(ii), rep->expr(val)));
+  };
+  static const auto dbg = [this] {
+    WARN("ignoring llvm.debug call.");
+    emit(Stmt::skip());
+  };
+
+  static const std::map<llvm::Intrinsic::ID, std::function<void()>> stmtMap {
     {llvm::Intrinsic::convert_from_fp16, f16UpCast},
-    {llvm::Intrinsic::convert_to_fp16, f16DownCast}
+    {llvm::Intrinsic::convert_to_fp16, f16DownCast},
+    //TODO: in future versions, there may be llvm.dbg.addr intrinisc
+    {llvm::Intrinsic::dbg_declare, dbg},
+    {llvm::Intrinsic::dbg_value, dbg},
+    {llvm::Intrinsic::expect, expect}
   };
+
   auto it = stmtMap.find(ii->getIntrinsicID());
   if (it != stmtMap.end())
-    return emit(it->second());
+    it->second();
   else
-    return emit(rep->call(ii->getCalledFunction(), *ii));
+    emit(rep->call(ii->getCalledFunction(), *ii));
```

Comment:
Could we issue a warning here to indicate that we encountered an intrinsic we are not currently handling/modeling?